streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -251,6 +251,21 @@ public StreamTask(final TaskId id,
         }
     }
 
+    // used for testing
+    long getStreamTime() {
+        return partitionGroup.timestamp();
+    }
+
+    // used for testing
+    long getPartitionTime(final TopicPartition partition) {
+        return partitionGroup.getPartitionTimestamp(partition);
+    }
+
+    // used for testing
+    void resetTimes() {
+        partitionGroup.clear();
+    }
+
```

Comment:
Can we add those method to the end of the class -- we have already a "section" for methods that we only need for testing.